Lib/test/test_tempfile.py

```diff
@@ -1652,6 +1652,45 @@ def test_cleanup_with_symlink_to_a_directory(self):
                          "were deleted")
         d2.cleanup()
 
+    @os_helper.skip_unless_symlink
+    @os_helper.skip_unless_working_chmod
+    @unittest.skipUnless(os.chmod in os.supports_follow_symlinks, 'needs chmod follow_symlinks support')
```

Comment:
Does the test actually require this support to pass? It seems to me that it should pass with or without this support, but without the support it doesn't really test anything, since there would have been no bug there to begin with. Is that accurate?

If so I think you should remove this line and let the test run, because it validates that your handling of no-supports-follow-symlinks in the code changes is correct.

Reply:
You're right, that's a leftover from when I was still trying to provoke a real `PermissionError` in the test code instead of just faking one. Should work just fine now, let's see...